### PR TITLE
feat: added irrelevant tags option to report post modal

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- Add another option to `ReportPostModal`
- Extend `RadioItem` to support `afterElement` (needed to render tags selection below radio)
- Update report post mutation to support tags as per https://github.com/dailydotdev/daily-api/pull/1319

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1446 #done

### CODE REVIEW

File: packages/shared/src/components/modals/ReportPostModal.tsx [hunk 1]
```diff
@@ -1,25 +1,77 @@
...
+              onClick={() => {
+                if (isSelected) {
+                  setSelectedTags(selectedTags.filter((t) => t !== tag));
+                } else {
+                  setSelectedTags([...selectedTags, tag]);
+                }
+              }}
+            >
+              #{tag}
+            </Button>
```
Comment (capJavert) [inline packages/shared/src/components/modals/ReportPostModal.tsx:67]: From what I see Lee used the same in his PR, and `TagButton` seems specific to follow concept used in onboarding. Let me know if we want to adjust this.

Comment (rebelchris) [reply packages/shared/src/components/modals/ReportPostModal.tsx:67]: All good for now, maybe a future improvement we can do?
To have a global tag component that can or cannot have actions? 👀

For now this works.

**Review by rebelchris — APPROVED** [review]
Nothing to add on this one, looks great 👍

File: packages/shared/src/components/modals/ReportPostModal.tsx (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/components/modals/ReportPostModal.tsx:67]: 🤝

**Pull request merged**